### PR TITLE
Reorganize optional dependencies

### PR DESCRIPTION
**Pull request by BSchilperoort**
Cartopy wheels are available on pip since august 2023 ([see changes here](https://github.com/SciTools/cartopy/pull/2197)), so conda is not needed anymore. I have updated the docs to reflect this.

This PR also makes the visualization packages optional. Matplotlib is required to make any visualization, and cartopy/cmocean are completely optional (but the user will be warned).
It could make sense to make these required as well, but I'm not sure.

I did not add any tests, because the visualization module will be reworked soon (#322)

### CODE REVIEW

**Review by Peter9192 — APPROVED** [review]
Nice work, I have a few small suggestions

File: docs/installation.md [hunk 1]
```diff
@@ -95,16 +93,25 @@ pip install wam2layers
 pip install wam2layers=3.0.0-beta5
```
[the remark this answers is not in the document]
Comment (BSchilperoort) [reply docs/installation.md:93]: It is just an example on how to install a specific version. I don't think it matters a lot if it's the latest version or not.

File: docs/installation.md [hunk 2]
```diff
@@ -45,47 +49,41 @@ which python3
...
 
 ```
-# Create a fresh conda environment
-conda create --name wamenv -c conda-forge python=3.11 jupyterlab cartopy matplotlib
-
-# Activate the environment
-conda activate wamenv
+pip install wam2layers
+pip install wam2layers[viz]  # includes packages required for visualization
+pip install wam2layers[complete]  # include all optional packages (for development)
```
[the remark this answers is not in the document]
Comment (BSchilperoort) [reply docs/installation.md:61]: I guess I can move the [complete] part to the dev docs (as normal users wouldn't need to use it)

File: src/wam2layers/analysis/visualization.py [hunk 3]
```diff
@@ -14,24 +14,53 @@
...
-    """Import cartopy if it is available; else raise."""
-    from importlib import import_module
+MISSING_CARTOPY_MSG = (
+    "Cartopy is not available, plots may look different than expected.\n"
+    "To install the optional visualization packages do `pip install wam2layers[viz]`."
+)
+MISSING_CMOCEAN_MSG = (
+    "cmocean is not available, colors of plots may look different than expected.\n"
+    "To install the optional visualization packages do `pip install wam2layers[viz]`."
+)
```
[the remark this answers is not in the document]
Comment (BSchilperoort) [reply src/wam2layers/analysis/visualization.py:24]: Yeah that's a consideration we would have to make. Many bigger/standard packages have split off everything non-essential into optional dependencies. For example: xarray doesn't have netcdf4 or matplotlib as dependency.

But it is debatable if this is required for our users. It makes installation a little bit more complex. However, if there are users doing their preprocessing and tracking on HPC or unusual systems, cartopy might not work there. Then they can't use wam2layers at all.

cmocean can probably be non-optional (it's 100% python), unless we want to make matplotlib optional too.

**Comment by Peter9192** [comment]
Let's only make cartopy optional, and matplotlib and cmcolors default dependencies

**Comment by BSchilperoort** [comment]
Test results are different from the expected results... For preprocessing as well as backtracking.

I did not have the error locally, until I reinstalled on a clean virtual environment. My older conda environment still runs everything fine :face_with_spiral_eyes:

**Comment by BSchilperoort** [comment]
I have pinned the xarray version for now. See #350

**Pull request merged**